input/socket/inputsocket.go

```diff
@@ -152,6 +158,54 @@ func (i *InputConfig) Start(ctx context.Context, msgChan chan<- logevent.LogEven
 	return eg.Wait()
 }
 
+// handleUDPpacketMode receives UDP packets and sends it out as messages one packet at a time
+func (i *InputConfig) handleUDPpacketMode(ctx context.Context, conn net.PacketConn, msgChan chan<- logevent.LogEvent) error {
+	eg, ctx := errgroup.WithContext(ctx)
+	b := make([]byte, i.BufferSize) // make read buffer
+	logger := goglog.Logger
+
+	// code to handle cancellation as ReadFrom is blocking and could take some time to time out if there is no traffic
+	eg.Go(func() error {
+		<-ctx.Done()
+		conn.Close()
+		return nil
+	})
+
+	// code to process input packets
+	eg.Go(func() error {
+		for {
+			select {
+			case <-ctx.Done():
+				return nil
+			default:
+			}
+			n, addr, err := conn.ReadFrom(b)
+			// handle data
+			if n > 0 {
+				if n >= i.BufferSize {
+					const bufIncSize = 600
+					logger.Errorf("%v UDP receive buffers (%v bytes) too small, should be increased!", i.Address, i.BufferSize)
+					i.BufferSize += bufIncSize
+					b = make([]byte, i.BufferSize)
+				} else {
+					extras := make(map[string]interface{})
+					extras["host_ip"] = addr.String()
+					_, _ = i.Codec.Decode(ctx, b[:n], extras, []string{}, msgChan)
```

Comment:
ignore all decode errors?

Reply:
I based my code on the existing code. All other code paths also ignore the result from Decode. I only added _, _ to be explicit about me ignoring the result.

If I should handle this I am a bit unsure how to handle the result. Please advice. I also should handle the error in the other code paths as well?

As Decode returns (ok bool, err error) I have four possible results:

```text
ok err
======
fa nil message not sent, no error to explain why
fa err message not sent with error
tr nil everything is ok, normal behaviour
tr err message sent, with error
```
I could write some code to send a log message in case the message was not sent or there is an error?

Reply:
The method signature is more nuanced.
The real failure is ONLY when `err` is returned.  That's a clear indication that the processing of data input has failed, either with parsing or with some other, codec-specific, data processing/decoding.
If `ok` is false, that only means that a new `Event` struct was not sent to the `msgChan`.  This could happen IFF a particular Codec implementation does not do any `Event` decoding per-say, which results in a new `Event`, but rather extracts some of the data and either does some "pre-processing" or adds attributes to the `Context` object.  (Neither of those are common/typical cases, but theoretically they both could occur)

In your particular case, I would only check for `err != nil` as a check.